admin/client/App/screens/Item/actions.js

```diff
@@ -23,13 +23,26 @@ export function selectItem (itemId) {
  */
 export function loadItemData () {
 	return (dispatch, getState) => {
+		// Crete an empty object to reference a point in memory.
+		// Dispatch this reference to our redux store to hold on to as a 'loadingRef'.
+		const thisLoadRef = {};
 		dispatch({
 			type: LOAD_DATA,
+			loadingRef: thisLoadRef,
 		});
 		const state = getState();
 		const list = state.lists.currentList;
 		// Load a specific item with the utils/List.js helper
 		list.loadItem(state.item.id, { drilldown: true }, (err, itemData) => {
+			// Once this async request has fired this callback, check that
+			// the point in memory referenced by thisLoadRef is the same point in memory
+			// referenced by loadingRef in the redux store.
+
+			// If it is, then this is the latest request, and it is safe to resolve it normally.
+			// If it is not a reference the same point in memory however,
+			// this means that this request is NOT the latest fired request,
+			// and so we'll bail out of it early.
+			if (getState().item.loadingRef !== thisLoadRef) return;
```

Comment:
Immutable only works in Redux because it doesn't not work. The entire state is supposed to be JSON-streamable at all times, as well as not mutated, however in practice it still mostly works if those don't hold.

As for the overlapping async call, I use something like http://stackoverflow.com/documentation/javascript/231/promises/8138/caching-promises-prevent-multiple-requests-for-the-same-thing#t=20160724090628003 to make sure running requests are all coalesced in the same request.